[pr into dev] refactored match.c to work with simclist instead of gmodule

MoldyAppleJuice opened the pull request:
- changed Makefile and tests/Makefile to compile simclist
- changed definitions of functions in match.c, and one in mtrie.c
- changed imports of match.c and mtrie.c
- updated test_match.c and test_mtrie.c to pass tests

RuolinZheng08 commented:
Everything looks good! Maybe you could add the word position into the match struct before merging into dev so we can have a fully-working version of the match class? Thanks!

MoldyAppleJuice commented:
I updated the match struct to include the int position.

changed
- match.h struct definition
- match.h/.c added get_position
- test_match.c updated tests to use position, test for position
- mtrie.c updated calls to match
- test_mtrie.c updated test calling match
